src/shared/metadata/cgroup_path_resolver.cc

```diff
@@ -38,11 +41,21 @@ StatusOr<std::string> CGroupBasePath(std::string_view sysfs_path) {
 
     // Attempt assuming naming scheme #1.
     std::string base_path = absl::StrCat(sysfs_path, "/cgroup/", cgroup_dir);
+
     if (fs::Exists(base_path)) {
       return base_path;
     }
   }
 
+  std::string cgv2_base_path = absl::StrCat(sysfs_path, "/cgroup");
+  struct statfs info;
+  auto fs_status = statfs(cgv2_base_path.c_str(), &info);
+  bool cgroupv2 = (fs_status == 0) && (info.f_type == CGROUP2_SUPER_MAGIC);
+
+  if (cgroupv2 || FLAGS_force_cgroup2_mode) {
+    return cgv2_base_path;
+  }
+
```

Comment:
Can you leave a comment to eventually move this above the cgroup v1 check? Since it's looking at the fs_type, that seems to be a more reliable check.

For now, leave it as is, because it is less risky of a change to have it the way you have it.

Reply:
Fixed in https://github.com/pixie-io/pixie/pull/635/commits/1b93a45662f7268c29981872f1c541777e9ace1f